### PR TITLE
chore: add examples of how to configure

### PR DESCRIPTION
**Pull request by lafirest**
Fixes [EMQX-9982](https://emqx.atlassian.net/browse/EMQX-9982)

### <samp>🤖 Generated by Copilot at 7eb653e</samp>

This pull request refactors the configuration management of EMQX by removing unused code and adding multiple example files for different features. It also updates the documentation to reflect the new location and format of the example files. The example files are located in the `examples` directory and have the suffix `.conf.example`. They demonstrate how to configure various aspects of EMQX, such as cluster, dashboard, hook, file transfer, alarm, and delayed publish.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: examples/alarm.conf.example [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+##--------------------------------------------------------------------
+## Note: This is an example of how to configure this feature
+##       you should copy and paste the below data into the emqx.conf for working
+
+alarm {
+    ## The actions triggered when the alarm is activated
+    ## Type: Array of the below enum
+    ##   - log :: write the alarm to log
+    ##   - publish :: publish the alarm as an MQTT message to the system topics
+    actions = [log]
```
Comment (zhongwencool) [inline examples/alarm.conf.example:14]: the default actions is `[log,publish]`, Maybe we should set example equal default?

File: examples/broker.conf.example [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+broker {
+    ## Session locking strategy in a cluster
+    ## Type:
+    ##   - local :: only lock the session on the current node
+    ##   - leader :: select only one remote node to lock the session
+    ##   - quorum :: select some nodes to lock the session
+    ##   - all :: lock the session on all the nodes in the cluster
+    session_locking_strategy = quorum
+
```
Comment (zhongwencool) [inline examples/broker.conf.example:19]: the entire broker is hidden, and move shared_subscription_strategy to `mqtt` root key.

File: examples/cluster-with-k8s.conf.example [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    ## Remove disconnected nodes from the cluster after this interval
+    autoclean = 5m
+
+    ## If true, the node will try to heal network partitions automatically
+    autoheal = true
+
+    k8s {
+      ## Kubernetes API endpoint URL
+      apiserver = "http://10.110.111.204:8080"
```
Comment (zhongwencool) [inline examples/cluster-with-k8s.conf.example:27]: Replace apiserver's value as "https://kubernetes.default.svc:443"

File: examples/cluster-with-static.conf.example [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    core_nodes = ['emqx1@192.168.0.1', 'emqx2@192.168.0.2']
+
+    ## Remove disconnected nodes from the cluster after this interval
+    autoclean = 5m
+
+    ## If true, the node will try to heal network partitions automatically
+    autoheal = true
+
+    ## List EMQX node names in the static cluster
+    static.seeds = ['emqx1@192.168.0.1', 'emqx2@192.168.0.2']
```
Comment (zhongwencool) [inline examples/cluster-with-static.conf.example:26]: all `['emqx1@192.168.0.1', 'emqx2@192.168.0.2']` is bad  hocon format.
should be 
```
"emqx1@192.168.0.1,emqx2@192.168.0.2"
% or
["emqx1@192.168.0.1", "emqx2@192.168.0.2"]
```
don't support `'` in hocon.

File: examples/dashboard-with-http.conf.example [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+dashboard {
+    ## JWT token expiration time
+    token_expired_time = 60m
+
+    ## Support Cross-Origin Resource Sharing (CORS)
+    cors = false
+
+    listeners.http {
+        ## Whether to enable the listener
+        enable = true
```
Comment (zhongwencool) [inline examples/dashboard-with-http.conf.example:18]: remove `enable = true`, Now the listeners is disable when bind = 0.

File: examples/dashboard-with-https.conf.example [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+dashboard {
+    ## JWT token expiration time
+    token_expired_time = 60m
+
+    ## Support Cross-Origin Resource Sharing (CORS)
+    cors = false
+
+    listeners.https {
+        ## Whether to enable the listener
+        enable = true
```
Comment (zhongwencool) [inline examples/dashboard-with-https.conf.example:18]: remove enable = true, Now the listeners is disable when bind = 0.

File: examples/listeners.quic.conf.example [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    certfile = "data/certs/cert.pem"
+
+    ## PEM format private key file
+    keyfile = "data/certs/key.pem"
+
+    ## Enable or disable peer verification
+    verify = verify_none ## to verify_peer to enable
+
+    ## TLS cipher suite names
+    ciphers = ["TLS_AES_256_GCM_SHA384", "TLS_AES_128_GCM_SHA256", "TLS_CHACHA20_POLY1305_SHA256"]
```
Comment (zhongwencool) [inline examples/listeners.quic.conf.example:47]: the cacerfile and keyfile is moving to ssl_options, @qzhuyan could you help to provide a full configuration about quic on e5.1.0 ?

File: examples/listeners.ssl.conf.example [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+    ## PEM format private key file
+    keyfile = "data/certs/key.pem"
+
+    ## Enable or disable peer verification
+    verify = verify_none  ## use verify_peer to enable
+
+    ## if `verify' is ebabled, whit true, the connection fails if the client does not have a certificate to send
+    fail_if_no_peer_cert = false
+
```
Comment (zhongwencool) [inline examples/listeners.ssl.conf.example:34]: It's wrong, should in ssl_options, you can get all config with default 
`emqx:get_config([listeners,ss,default]).

File: examples/listeners.wss.conf.example [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    ## Enable or disable peer verification
+    verify = verify_none  ## use verify_peer to enable
+
+    ## if `verify' is ebabled, whit true, the connection fails if the client does not have a certificate to send
+    fail_if_no_peer_cert = false
+
+    ## Enable TLS session reuse
+    reuse_sessions = true
+
+    ## Maximum number of non-self-issued intermediate certificates that can follow the peer certificate in a valid certification path
```
Comment (zhongwencool) [inline examples/listeners.wss.conf.example:36]: all listener's ssl_options should be in `ssl_options{}`.

File: examples/rpc-with-ssl.conf.example [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+    ## Path to TLS certificate file used to validate identity of the cluster nodes
+    certfile = "data/cert.pem"
+
+    ## Path to the private key file
+    keyfile = "data/key.pem"
+
+    ## Path to certification authority TLS certificate file
+    cacertfile = "data/cacert.pem"
+
```
Comment (zhongwencool) [inline examples/rpc-with-ssl.conf.example:49]: We already hide rpc key, Maybe we should remove the entire rpc example, Add it back if necessary in the future.

**Review by zhongwencool — DISMISSED** [review]

File: examples/listeners.quic.conf.example
Comment (cherryxkz) [inline examples/listeners.quic.conf.example:1]: ssl_options {
        cacertfile = "data/certs/listeners/quic/test/cacert-1D7CF2F305E31CD1"
        certfile = "data/certs/listeners/quic/test/cert-9ABB012C4B9589A6"
        depth = 10
        fail_if_no_peer_cert = false
        keyfile = "data/certs/listeners/quic/test/key-85262C1E70624A26"
        verify = "verify_peer"
        versions = ["tlsv1.3", "tlsv1.2", "tlsv1.1", "tlsv1"]
      }

File: examples/file_transfer-with-s3-exporter.conf.example [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+                bucket = "my-bucket"
+
+                ## TTL of file download URLs exposed through File Transfer API
+                url_expire_time = 1h
+
+                ## Enable the HTTPS
+                transport_options.ssl.enable = true
+
+                ## Timeout for connection attempts
+                connect_timeout = 15s
```
Comment (Le9Wang) [inline examples/file_transfer-with-s3-exporter.conf.example:63]: It should be transport_options.connect_timeout

File: examples/file_transfer-with-s3-exporter.conf.example [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+                ## Attempt to talk through IPv6 first
+                ipv6_probe = true
```
Comment (Le9Wang) [inline examples/file_transfer-with-s3-exporter.conf.example:66]: It should be: transport_options.ipv6_probe = true

File: examples/flapping_detect.conf.example [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+## Flapping Detect
+##
+## Ban the client when the times of connections exceed the limit in the time window
+##--------------------------------------------------------------------
+## Note: This is an example of how to configure this feature
+##       you should copy and paste the below data into the emqx.conf for working
+
+flapping_detect {
+    ## Feature switch
+    enable = false
```
Comment (Le9Wang) [inline examples/flapping_detect.conf.example:11]: It might be: enable = true

File: examples/listeners.quic.conf.example
Comment (cherryxkz) [inline examples/listeners.quic.conf.example:1]: ssl_options {
         ## Trusted PEM format CA certificates bundle file
         cacertfile = "${EMQX_ETC_DIR}/certs/cacert.pem"

         ## PEM format certificates chain file
         certfile = "${EMQX_ETC_DIR}/certs/cert.pem"

         ## PEM format private key file
         keyfile = "${EMQX_ETC_DIR}/certs/key.pem"

         ## Enable or disable peer verification
         verify = verify_peer ## to verify_peer to enable
    
     }

File: examples/listeners.wss.conf.example
Comment (cherryxkz) [inline examples/listeners.wss.conf.example:1]: ssl_options {
     ## Trusted PEM format CA certificates bundle file
     cacertfile = "${EMQX_ETC_DIR}/certs/cacert.pem"

     ## PEM format certificates chain file
     certfile = "${EMQX_ETC_DIR}/certs/cert.pem"

     ## PEM format private key file
     keyfile = "${EMQX_ETC_DIR}/certs/key.pem"

     

 }

File: examples/prometheus.conf.example [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+    }
+
+    ## Job Name that is pushed to the Push Gateway.
+    ## Available variable:
+    ## - ${name}: Name of EMQX node
+    ## - ${host}: Host name of EMQX node
+    job_name = "${name}/instance/${name}~${host}"
+
+    ## set true to enable this
+    enable = false
```
Comment (Le9Wang) [inline examples/prometheus.conf.example:27]: It might be: enable = true

File: examples/psk_authentication.conf.example [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+## Pre-Shared Keys authentication
+##
+## Config to enable TLS-PSK authentication
+##--------------------------------------------------------------------
+## Note: This is an example of how to configure this feature
+##       you should copy and paste the below data into the emqx.conf for working
+
+psk_authentication {
+    ## set true to enable TLS PSK support
+    enable = false
```
Comment (Le9Wang) [inline examples/psk_authentication.conf.example:11]: It might be: enable = true

**Review by HJianBo — APPROVED** [review]

**Pull request merged**